### PR TITLE
PR: Generalice previous is_instance test to search other avoided patterns

### LINKED ISSUE #4161: Create a test for finding print statements in the code
Very often PRs leave a print statement behind, as part of the initial testing. A test that looks for print statements and fails if it does find it would be very helpful.

@rlaverde we could generalize the regex check you did, to find generic code snippets. What do you think?

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes #4161.

### CODE REVIEW

File: spyder/utils/introspection/plugin_server.py [hunk 1]
```diff
@@ -120,10 +120,11 @@ def initialize(self, plugin_name):
 
 
 if __name__ == '__main__':
+    from spyder.config.base import debug_print
     args = sys.argv[1:]
     if not len(args) == 2:
-        print('Usage: plugin_server.py client_port plugin_name')
+        debug_print('Usage: plugin_server.py client_port plugin_name')
```
Comment (ccordoba12) [inline spyder/utils/introspection/plugin_server.py]: Please use `# spyder: test-skip` here

File: spyder/utils/introspection/plugin_server.py [hunk 2]
```diff
@@ -120,10 +120,11 @@ def initialize(self, plugin_name):
...
         sys.exit(0)
     plugin = PluginServer(*args)
-    print('Started')
+    debug_print('Started')
```
Comment (ccordoba12) [inline spyder/utils/introspection/plugin_server.py]: `# spyder: test-skip`

File: spyder/widgets/pathmanager.py [hunk 3]
```diff
@@ -244,11 +244,12 @@ def add_path(self):
 def test():
     """Run path manager test"""
     from spyder.utils.qthelpers import qapplication
+    from spyder.config.base import debug_print
     _app = qapplication()  # analysis:ignore
     test = PathManager(None, pathlist=sys.path[:-10],
                        ro_pathlist=sys.path[-10:])
     test.exec_()
-    print(test.get_path_list())
+    debug_print(test.get_path_list())
```
Comment (ccordoba12) [inline spyder/widgets/pathmanager.py]: `# spyder: test-skip`

File: spyder/widgets/variableexplorer/importwizard.py [hunk 4]
```diff
@@ -632,10 +632,11 @@ def process(self):
 def test(text):
     """Test"""
     from spyder.utils.qthelpers import qapplication
+    from spyder.config.base import debug_print
     _app = qapplication()  # analysis:ignore
     dialog = ImportWizard(None, text)
     if dialog.exec_():
-        print(dialog.get_data())
+        debug_print(dialog.get_data())
```
Comment (ccordoba12) [inline spyder/widgets/variableexplorer/importwizard.py]: `# spyder: test-skip`

File: spyder/widgets/variableexplorer/objecteditor.py [hunk 5]
```diff
@@ -167,11 +165,11 @@ def __init__(self):
...
-    print(oedit(foobar))
-    print(oedit(example))
-    print(oedit(np.random.rand(10, 10)))
-    print(oedit(oedit.__doc__))
-    print(example)
+    debug_print(oedit(foobar))
+    debug_print(oedit(example))
+    debug_print(oedit(np.random.rand(10, 10)))
+    debug_print(oedit(oedit.__doc__))
+    debug_print(example)
```
Comment (ccordoba12) [inline spyder/widgets/variableexplorer/objecteditor.py]: `# spyder: test-skip` for all of these

File: spyder/widgets/formlayout.py [hunk 6]
```diff
@@ -577,19 +578,19 @@ def create_datagroup_example():
     #--------- datalist example
     datalist = create_datalist_example()
     def apply_test(data):
-        print("data:", data)
-    print("result:", fedit(datalist, title="Example",
+        debug_print("data:", data)
```
Comment (ccordoba12) [inline spyder/widgets/formlayout.py]: `# spyder: test-skip`

File: spyder/widgets/formlayout.py [hunk 7]
```diff
@@ -577,19 +578,19 @@ def create_datagroup_example():
...
+    debug_print("result:", fedit(datalist, title="Example",
```
Comment (ccordoba12) [inline spyder/widgets/formlayout.py]: `# spyder: test-skip`

File: spyder/widgets/formlayout.py [hunk 8]
```diff
@@ -577,19 +578,19 @@ def create_datagroup_example():
...
                            comment="This is just an <b>example</b>.",
                            apply=apply_test))
     
     #--------- datagroup example
     datagroup = create_datagroup_example()
-    print("result:", fedit(datagroup, "Global title"))
+    debug_print("result:", fedit(datagroup, "Global title"))
```
Comment (ccordoba12) [inline spyder/widgets/formlayout.py]: `# spyder: test-skip`

File: spyder/widgets/formlayout.py [hunk 9]
```diff
@@ -577,19 +578,19 @@ def create_datagroup_example():
...
     
     #--------- datagroup inside a datagroup example
     datalist = create_datalist_example()
     datagroup = create_datagroup_example()
-    print("result:", fedit(((datagroup, "Title 1", "Tab 1 comment"),
+    debug_print("result:", fedit(((datagroup, "Title 1", "Tab 1 comment"),
```
Comment (ccordoba12) [inline spyder/widgets/formlayout.py]: `# spyder: test-skip`

File: spyder/utils/vcs.py [hunk 10]
```diff
@@ -146,8 +143,9 @@ def get_git_revision(repopath):
...
-    print(get_vcs_root(osp.dirname(__file__)))
-    print(get_vcs_root(r'D:\Python\ipython\IPython\kernel'))
+    from spyder.config.base import debug_print
+    debug_print(get_vcs_root(osp.dirname(__file__)))
+    debug_print(get_vcs_root(r'D:\Python\ipython\IPython\kernel'))
     #run_vcs_tool(r'D:\Python\userconfig\userconfig', 'commit')
-    print(get_git_revision(osp.dirname(__file__)+"/../.."))
-    print(get_git_revision('/'))
+    debug_print(get_git_revision(osp.dirname(__file__)+"/../.."))
+    debug_print(get_git_revision('/'))
```
Comment (ccordoba12) [inline spyder/utils/vcs.py]: Change to `# spyder: test-skip` all of these

File: spyder/utils/system.py [hunk 11]
```diff
@@ -62,11 +62,12 @@ def psutil_phymem_usage():
...
+    debug_print(memory_usage())
     if os.name == 'nt':
         #  windll can only be imported if os.name = 'nt' or 'ce'
-        print("*"*80)
-        print(windows_memory_usage.__doc__)
-        print(windows_memory_usage())
\ No newline at end of file
+        debug_print("*"*80)
+        debug_print(windows_memory_usage.__doc__)
+        debug_print(windows_memory_usage())
```
Comment (ccordoba12) [inline spyder/utils/system.py]: `# spyder: test-skip` for all of these

File: spyder/utils/dochelpers.py [hunk 12]
```diff
@@ -328,17 +326,19 @@ def isdefined(obj, force_import=False, namespace=None):
...
+    debug_print(getargtxt(Test.method))
+    debug_print(isdefined('numpy.take', force_import=True))
+    debug_print(isdefined('__import__'))
+    debug_print(isdefined('.keys', force_import=True))
+    debug_print(getobj('globals'))
+    debug_print(getobj('globals().keys'))
+    debug_print(getobj('+scipy.signal.'))
+    debug_print(getobj('4.'))
+    debug_print(getdoc(sorted))
+    debug_print(getargtxt(sorted))
```
Comment (ccordoba12) [inline spyder/utils/dochelpers.py]: `# spyder: test-skip` for all of these

File: spyder/plugins/shortcuts.py [hunk 13]
```diff
@@ -835,7 +835,7 @@ def test():
     table = ShortcutsTable()
     table.show()
     app.exec_()
-    print([str(s) for s in table.source_model.shortcuts])
+    debug_print([str(s) for s in table.source_model.shortcuts])
```
Comment (ccordoba12) [inline spyder/plugins/shortcuts.py]: `# spyder: test-skip`

**Comment by ccordoba12** [comment]
The rationale of my suggested changes is this:

* print's at the end of files should be left as print's because they are needed to easily debug/develop a given feature or widget.
* print's that are needed to show some internal info a widget/module should be moved to debug_print's

**Comment by ccordoba12** [comment]
@rlaverde, please finish this one so we can merge it :-)

**Comment by rlaverde** [comment]
> please finish this one so we can merge it :-)

Done 😄

**Comment by ccordoba12** [comment]
Thanks @rlaverde! Looks good now :-)

**Pull request merged**